File: api-stockhub/stockhub/src/main/java/com/api/stockhub/repositories/ColaboradorRepository.java
package com.api.stockhub.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.api.stockhub.models.ColaboradorModel;

@Repository
public interface ColaboradorRepository extends JpaRepository<ColaboradorModel, Long>{

}

File: Api/Api/Infrastructure/Repositories/ColaboradorRepository.cs
using Api.Domain.Interfaces;
using Api.Domain.Models;
using Api.Infrastructure.Context;
using Microsoft.EntityFrameworkCore;

namespace Api.Infrastructure.Repositories
{
    public class ColaboradorRepository : BaseRepository, IColaboradorRepository
    {
        private readonly ApplicationDbContext _context;
        public ColaboradorRepository(ApplicationDbContext context) : base(context) 
        {
            this._context = context;
        }


        public async Task<IEnumerable<ColaboradorModel>> GetColaboradorAsync()
        {
            return await _context.Colaboradores.Include(x => x.Movimentacao).ToListAsync();
        }

        public async Task<ColaboradorModel> GetColaboradorByIdAsync(long id)
        {
            return await _context.Colaboradores.Include(x => x.Movimentacao).Where(x => x.Id == id).FirstOrDefaultAsync();
        }
    }
}
